File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int i = 0, j = 0;
        while (i < str1.length() && j < str2.length()){
            if (str1.charAt(i) == str2.charAt(j) || (str1.charAt(i) - 'a'  + 1) % 26 == str2.charAt(j) - 'a')
                j++;
            i++;
        }
        return str2.length() == j ? true : false;
    }
}

File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        i, j = 0, 0
        while i < len(str1) and j < len(str2):
            if str1[i] == str2[j] or (ord(str1[i]) + 1 - ord('a')) % 26  == ord(str2[j]) - ord('a'):
                j += 1
            i += 1
        if j != len(str2):
            return False
        return True